### ISSUE TITLE
Robot occasionally get stuck in clinic world 

### ISSUE THREAD

**Issue by tanyouliang95** [open-rmf/rmf_demos#45]
Once a while, the `deliveryRobot` will get stuck in front of a door in `clinic.world`. This might due to some waypoint estimation in slotcar. This is not highly reproducible in every test. Create this issue ticket for tracking.

https://user-images.githubusercontent.com/32189404/118443853-fdce7500-b71e-11eb-8fb2-4692460b3d46.mp4

**Comment by mxgrey** [comment]
I can't view the video because Firefox is saying it's corrupt, but I think I've experienced the issue you're describing.

I believe there are certain door-approaching waypoints in the clinic that are too close to the door that they're approaching. As a result, the slotcar's very primitive collision avoidance system is being triggered by the door being closed, and so the slotcar is stopping before it reaches the waypoint that it needs to get to. Also the door won't open until the robot reaches that waypoint. The end result is a deadlock where the door won't open and the slotcar won't approach.

**Comment by tanyouliang95** [comment]
Reuploaded with a gif. Yes, previously some of the waypoints are too close to the doors, and I have shifted those which are too close to the walls/doors.

**Comment by mxgrey** [comment]
Ah, in the gif I think the issue may be with the replanning interrupt. This is probably the same issue as [this one](https://github.com/osrf/rmf_core/issues/146) which will hopefully be resolved by eventually moving the slotcar over to be controlled by a future version of `free_fleet`.

**Comment by tanyouliang95** [comment]
Yes you are right. I observed this multiple times. In this scenario, the robot was interrupted, also causing it to skip the intended `OpenDoor` phase after replanning from a different Start waypoint. Which this also explains why the door remains close.
